### ISSUE TITLE
attribute is missing

### ISSUE THREAD

**Issue by hehengkaifj** [xmlet/XsdParser#25]
<xsd:element name="AgeRecommendation">
		<xsd:complexType>
			<xsd:sequence>
				<xsd:element name="MinimumManufacturerAgeRecommended" type="MinimumAgeRecommendedDimension" minOccurs="0"/>
			</xsd:sequence>
		</xsd:complexType>
	</xsd:element>

<xsd:complexType name="MinimumAgeRecommendedDimension">
		<xsd:simpleContent>
			<xsd:extension base="xsd:nonNegativeInteger">
				<xsd:attribute name="unitOfMeasure" type="AgeRecommendedUnitOfMeasure" use="required"/>
			</xsd:extension>
		</xsd:simpleContent>
	</xsd:complexType>

<xsd:simpleType name="AgeRecommendedUnitOfMeasure">
		<xsd:restriction base="xsd:string">
			<xsd:enumeration value="months"/>
			<xsd:enumeration value="years"/>
		</xsd:restriction>
	</xsd:simpleType>

i can not find the attribute name "unitOfMeasure" in this case

**Comment by hehengkaifj** [comment]
@lcduarte

**Comment by lcduarte** [comment]
Hello,

I've tested with that XSD and somehow the attribute is not present in the XsdExtension element. I'll need some time to understand the issue, I'll probably get back to you on the weekend.

**Comment by hehengkaifj** [comment]
btw... the element "MinimumManufacturerAgeRecommended" is also missing when it is a base xsd 
eg:
<xsd:include schemaLocation="base.xsd"/> 
<xsd:element ref="AgeRecommendation" minOccurs="0"/>

**Comment by lcduarte** [comment]
Can you send the xsd files that you are using so I can reproduce the issue more accurately?

**Comment by hehengkaifj** [comment]
[https://images-na.ssl-images-amazon.com/images/G/01/rainier/help/xsd/release_1_9/ToysBaby._TTH_.xsd](url)
[https://images-na.ssl-images-amazon.com/images/G/01/rainier/help/xsd/release_1_9/amzn-base._TTH_.xsd](url)

**Comment by lcduarte** [comment]
I found the problem and I've solved both situations reported. I'm trying to perform a new release but I'm getting some kind of timeout at the moment. I'll try it later and inform you.

**Comment by hehengkaifj** [comment]
thanks for the fast response!
waiting for the new release

**Comment by hehengkaifj** [comment]
hello
can you give me a demo about how to deal with ref attribute ,it seems like not a correct type of UnsolvedReference in the wiki
here is XSD files
https://images-na.ssl-images-amazon.com/images/G/01/rainier/help/xsd/release_1_9/Books.xsd

**Comment by lcduarte** [comment]
I think my deployment "failed" but the new version is already deployed, ([1.0.30](https://github.com/xmlet/XsdParser/tree/xsdParser-1.0.30)).

Simply put, the ```XsdParser``` has two different iterations. The first iteration goes through all the xsd element nodes and either converts them into a ```ConcreteElement```/```NamedConcreteElement``` or ```UnsolvedReference```.

* ```ConcreteElements``` are elements which don't have a ```name``` attribute, nor a ```ref``` attribute.
* ```NamedConcreteElements``` are elements which have a ```name``` attribute.
* ```UnsolvedReferences``` are elements which have the ```ref``` attribute.

In the second iteration the Parser goes through all the ```UnsolvedReferences``` and tries to find the corresponding ```NamedConcreteElement```. In concrete terms it tries to find the xsd element with the same ```name``` as the name present in the ```ref``` attribute of another element. 

When a ```ConcreteElement``` matches with a ```UnsolvedReference``` the ```UnsolvedReference``` object is replaced by a **clone** of the original ```ConcreteElement```. In the latest version I improved the cloning process which resulted in the bugs reported in this issue.

Any ```UnsolvedReferences``` present after the parsing process means that the Parser couldn't find the element with the ```name``` attribute with the same value as the ```ref``` attribute of the element wrapped in the ```UnsolvedReference``` object. 

Either that or something is wrong with the Parser 😄

**Comment by hehengkaifj** [comment]
i found another problem when parsing the element path  "AutoAccessory/ProductType/AutoAccessoryMisc/Amperage".
the element of Amperage is exist, but after parsed its type is ' UnsolvedReference' 
here is the xsd file
https://images-na.ssl-images-amazon.com/images/G/01/rainier/help/xsd/release_4_1/AutoAccessory.xsd

**Comment by lcduarte** [comment]
I've investigated the issue and it looks like the problem is due to it being a "second degree" reference. 

This means that since the ```AutoAccessoryMisc``` is referenced on Product Type and then the ```AutoAccessoryMisc``` itself has its own "references" such as the type ```AmperageDimension``` the second level reference won't be resolved. 

I'll need some time to solve this.

**Comment by lfrancke** [comment]
It looks like I'm hitting the same issue with 1.0.32.
I tried with this example XSD:

```
<?xml version="1.0" encoding="UTF-8"?>
<xs:schema targetNamespace="http://foo" xmlns="http://foo" xmlns:xs="http://www.w3.org/2001/XMLSchema">
  <xs:element name="A" type="AType"/>
  <xs:complexType name="AType">
    <xs:simpleContent>
      <xs:extension base="xs:string">
        <xs:attribute name="AttributeA" type="xs:string" use="optional"/>
        <xs:attribute name="AttributeB" type="xs:string" use="optional"/>
      </xs:extension>
    </xs:simpleContent>
  </xs:complexType>
</xs:schema>
```

I cannot find a way to get to those attributes.

**Comment by lcduarte** [comment]
Hello,

First, sorry for taking so long solving this one @hehengkaifj , it really was a tough one. Like I said before XsdParser was only solving first degree references, for example:

```xml
<xsd:element name="A" ref="B"/>

<xsd:element name="B" ref="C"/>

<xsd:element name="C"/>
```

In this case XsdParser would be able to resolve both references.

* Element A would have a clone of B;
* Element B would have a clone of C;

But, the B clone that element A had wouldn't have the reference resolved with element C.

This behavior was solved by adding the unsolved references of the clone in the Parser Reference Solving process. This way all clones should have be similar and have the same references solved.

I released a new version, ([1.1.0](https://github.com/xmlet/XsdParser/tree/xsdParser-1.1.0)), with these changes, hopefully those solve your issues!

If possible give me some feedback.
